Shell verifier didn't provide KITCHEN_USERNAME

## Description

I use Testinfra and the shell verifier not set KITCHEN_USERNAME environment variable

## Kitchen Version

Test Kitchen version 1.20.0

## Ruby Version

```ruby
ruby 2.3.3p222 (2016-11-21) [x86_64-linux-gnu]
```

## Platform Version

Debian 9

## Replication Case

My Kitchen configuration
```
# vim: ft=yaml
---
driver:
  name: openstack
  xxxxx

transport:
  ssh_key: <%= ENV['HOME'] + '/.ssh/id_rsa' %>

platforms:
  - name: debian9
    driver:
      image_ref: "Debian 9"
    transport:
      username: debian
  - name: centos7
    driver:
      image_ref: "Centos 7"
    transport:
      username: centos

provisioner:
  name: salt_solo
  is_file_root: true

verifier:
  name: shell
  command: py.test -v --color=yes --host="ssh://${KITCHEN_USERNAME}@${KITCHEN_HOSTNAME}?ssh_identity_file=${KITCHEN_SSH_KEY}" --junit-xml "junit-${KITCHEN_INSTANCE}.xml" test/integration/${KITCHEN_SUITE}*
```



## Kitchen Output

Kitchen can create the instance and connect to it for converge. The `username` set in platform is correctly used, but when Testinfra try to connect, the current user is used instead.

I've add `env |grep KIT && ` before the testinfra call in my verifier command, and I didn't see `KITCHEN_USERNAME`

```ruby
-----> Starting Kitchen (v1.20.0)
-----> Creating <default-debian9>...
       OpenStack instance with ID of <260691da-15ce-4840-83e7-45dbd6d5359c> is ready.
       [.......]
       [SSH] Established
       Finished creating <default-debian9> (0m53.59s).
-----> Converging <default-debian9>...
       [.......]
       Finished converging <default-debian9> (2m2.00s).
-----> Setting up <default-debian9>...
       Finished setting up <default-debian9> (0m0.00s).
-----> Verifying <default-debian9>...
       [Shell] Verify on instance default-debian9 with state={:server_id=>"260691da-15ce-4840-83e7-45dbd6d5359c", :hostname=>"x.x.x.x", :last_action=>"setup", :last_error=>nil}

KITCHEN_HOSTNAME=x.x.x.x
KITCHEN_PLATFORM=debian9
KITCHEN_FLAVOR=small
KITCHEN_LAST_ACTION=setup
KITCHEN_YAML=/opt/kitchen/kitchen.openstack.yml
KITCHEN_NETWORK_PUBLIC=Ext-Net
KITCHEN_SUITE=default
KITCHEN_SERVER_ID=260691da-15ce-4840-83e7-45dbd6d5359c
KITCHEN_INSTANCE=default-debian9
KITCHEN_SSH_KEY_NAME=kitchen-test
KITCHEN_SERVER_PREFIX=kitchen-test
KITCHEN_LAST_ERROR=

============================= test session starts ==============================
platform linux2 -- Python 2.7.13, pytest-3.5.0, py-1.5.3, pluggy-0.6.0 -- /usr/bin/python
plugins: testinfra-1.11.1
[.......]
=================================== FAILURES ===================================
INFO:testinfra:RUN CommandResult(command="ssh x.x.x.x 'uname -r'", exit_status=0, stdout='Please login as the user "debian" rather than the user "root".\n\n', stderr=None)
```

## Kitchen Diagnose
I see `KITCHEN_USERNAME` in the PR https://github.com/test-kitchen/test-kitchen/pull/741 , but I can't found it in the code.

## NOTE:

This repository is for core issues with the Test-Kitchen framework, so if the issue is with a plugin you may be referred to file the issue with the appropriate plugin.

--- a/lib/kitchen/verifier/shell.rb
+++ b/lib/kitchen/verifier/shell.rb
@@ -88,6 +88,7 @@ module Kitchen
         env_state[:environment]["KITCHEN_INSTANCE"] = instance.name
         env_state[:environment]["KITCHEN_PLATFORM"] = instance.platform.name
         env_state[:environment]["KITCHEN_SUITE"] = instance.suite.name
+        env_state[:environment]["KITCHEN_USERNAME"] = instance.transport[:username] if instance.transport.key?(:username)
         state.each_pair do |key, value|
           env_state[:environment]["KITCHEN_" + key.to_s.upcase] = value.to_s
         end

--- a/lib/kitchen/verifier/shell.rb
+++ b/lib/kitchen/verifier/shell.rb
@@ -88,7 +88,7 @@ module Kitchen
         env_state[:environment]["KITCHEN_INSTANCE"] = instance.name
         env_state[:environment]["KITCHEN_PLATFORM"] = instance.platform.name
         env_state[:environment]["KITCHEN_SUITE"] = instance.suite.name
-        env_state[:environment]["KITCHEN_USERNAME"] = instance.transport[:username] if instance.transport.key?(:username)
+        env_state[:environment]["KITCHEN_USERNAME"] = instance.transport[:username] if instance.respond_to?(:transport)
         state.each_pair do |key, value|
           env_state[:environment]["KITCHEN_" + key.to_s.upcase] = value.to_s
         end

--- a/spec/kitchen/verifier/shell_spec.rb
+++ b/spec/kitchen/verifier/shell_spec.rb
@@ -34,13 +34,21 @@ describe Kitchen::Verifier::Shell do
     { test_base_path: "/basist", kitchen_root: "/rooty" }
   end
 
+  let(:transport) do
+    t = mock("transport")
+    t.responds_like_instance_of(Kitchen::Transport::Ssh)
+    t.stubs(:[])
+    t
+  end
+
   let(:instance) do
     stub(
       name: [platform.name, suite.name].join("-"),
       to_str: "instance",
       logger: logger,
       suite: suite,
-      platform: platform
+      platform: platform,
+      transport: transport
     )
   end
 
@@ -90,6 +98,13 @@ describe Kitchen::Verifier::Shell do
         config[:shellout_opts][:environment]["KITCHEN_INSTANCE"].must_equal "coolbeans-fries"
         config[:shellout_opts][:environment]["KITCHEN_PLATFORM"].must_equal "coolbeans"
         config[:shellout_opts][:environment]["KITCHEN_SUITE"].must_equal "fries"
+        config[:shellout_opts][:environment]["KITCHEN_USERNAME"].must_be_nil
+      end
+
+      it "transport username is set to environment" do
+        transport.stubs(:[]).with(:username).returns("demigod")
+        verifier.call(state)
+        config[:shellout_opts][:environment]["KITCHEN_USERNAME"].must_equal "demigod"
       end
 
       it "raises ActionFailed if set false to :command" do
@@ -109,6 +124,7 @@ describe Kitchen::Verifier::Shell do
       let(:transport) do
         t = mock("transport")
         t.responds_like_instance_of(Kitchen::Transport::Ssh)
+        t.stubs(:[])
         t
       end
 

